pkg/cli/copy.go

```diff
@@ -248,13 +264,24 @@ func (a *ImageCopy) copyRepo(args []string, sourceOpts, destOpts []remote.Option
 }
 
 func (a *ImageCopy) copyTag(source name.Reference, newTag string, sourceOpts []remote.Option) error {
+	// -a is not supported for this operation, so check if it is set and return an error if so
+	if a.AllTags {
+		return errors.New("cannot use --all-tags with a tag destination")
+	}
+
 	sourceIndex, err := remote.Index(source, sourceOpts...)
 	if err != nil {
 		return err
 	}
 
 	dest := source.Context().Tag(newTag)
 
+	// Parse it again to make sure that the tag provided by the user is valid
+	_, err = name.ParseReference(dest.String())
+	if err != nil {
```

Comment:
nit: you can condense this to a single line

```suggestion
	if _, err = name.ParseReference(dest.String()); err != nil {
```